backend/src/handle_socket.rs

```diff
@@ -370,14 +370,24 @@ async fn handle_client_message(
         socket.send(resp).await?;
         return Ok(None);
     };
-    let Ok(ClientMessage {
-        request_id,
-        message: Some(msg),
-    }) = ClientMessage::decode(Bytes::from(msg))
-    else {
-        let resp = request_failed(String::new(), "Unknown", "Expected Client message");
-        socket.send(resp).await?;
-        return Ok(None);
+    let (request_id, msg) = match ClientMessage::decode(Bytes::from(msg)) {
+        Ok(cli_msg) => (cli_msg.request_id, match
+            cli_msg.message {
+                Some(message) => message,
+                None => {
+                    eprintln!("Error: Failed to decode client message: {:?}", "no error");
+                    let resp = request_failed(String::new(), "Unknown", "Expected Client message");
+                    socket.send(resp).await?;
+                    return Ok(None);
+                }
+            }
+        ),
+        Err(error) => {
+            eprintln!("Error: Failed to decode client message: {:?}", error);
+            let resp = request_failed(String::new(), "Unknown", "Expected Client message");
+            socket.send(resp).await?;
+            return Ok(None);
+        }
```

Comment:
This change seems unnecessary, the request_failed function already does a `tracing::error`